docs/src/apimanual.md

```diff
@@ -410,31 +406,67 @@ See also the attributes [`ConstraintPrimal`](@ref), and
+| Proved infeasible                       | `Infeasible`          | 1               | `NoSolution`                             | `InfeasibilityCertificate`               |
+| Optimal within relaxed tolerances       | `AlmostOptimal`       | 1               | `FeasiblePoint` or `AlmostFeasiblePoint` | `FeasiblePoint` or `AlmostFeasiblePoint` |
+| Detected an unbounded ray of the primal | `DualInfeasible`      | 1               | `InfeasibilityCertificate`               | `NoSolution`                             |
+| Stall                                   | `SlowProgress`        | 1               | ?                                        | ?                                        |
 
-When a global solver returns `Success` and the primal result is a
-`FeasiblePoint`, then it is implied that the primal result is indeed a globally
-optimal solution up to the specified tolerances. Typically, no dual certificate
-is available to certify optimality. The [`ObjectiveBound`](@ref) should provide
-extra information on the optimality gap.
+#### Global branch-and-bound solvers
 
-#### Local solver
+Mixed-integer programming solvers fall into this category.
 
-For solvers which perform a search based only on local criteria (for example, gradient descent), without additional knowledge of the structure of the problem, we can say only that `Success` and `FeasiblePoint` imply that the primal result belongs to the class of points which the chosen algorithm is known to converge to. Gradient descent algorithms may converge to saddle points, for example. It is also possible for such algorithms to converge to infeasible points, in which case the termination status would be `Success` and the primal result status would be `InfeasiblePoint`. This does not imply that the problem is infeasible and so cannot be called a certificate of infeasibility.
+| What happened?                                   | `TerminationStatus()`   | `ResultCount()` | `PrimalStatus()`  | `DualStatus()` |
+| ------------------------------------------------ | ----------------------- | --------------- | ----------------- | -------------- |
+| Proved optimality                                | `Optimal`               | 1               | `FeasiblePoint`   | `NoSolution`   |
+| Presolve detected infeasibility or unboundedness | `InfeasibleOrUnbounded` | 0               | `NoSolution`      | `NoSolution`   |
+| Proved infeasibility                             | `Infeasible`            | 0               | `NoSolution`      | `NoSolution`   |
+| Timed out (no solution)                          | `TimeLimit`             | 0               | `NoSolution`      | `NoSolution`   |
+| Timed out (with a solution)                      | `TimeLimit`             | 1               | `FeasiblePoint`   | `NoSolution`   |
+| `CPXMIP_OPTIMAL_INFEAS`                          | `AlmostOptimal`         | 1               | `InfeasiblePoint` | `NoSolution`   |
```

Comment:
I can see someone reading this and thinking, didn't it find an `AlmostFeasiblePoint`? The answer is "it depends" but it probably goes nicely with `AlmostOptimal`.

Reply:
I almost put `Optimal` | `InfeasiblePoint` for this case, because that's what the solver actually says. I stepped back because you could make an argument that it would be easy to see `Optimal` and forget to check the `PrimalStatus` and miss the issue. `AlmostOptimal` and `AlmostFeasiblePoint` aren't strictly correct because the solver never reported that relaxed tolerances are satisfied. Not sure...

Reply:
I mean, the [docs say](https://www.ibm.com/support/knowledgecenter/SSSA5P_12.6.3/ilog.odms.cplex.help/refcallablelibrary/macros/CPXMIP_OPTIMAL_INFEAS.html) "Problem is optimal with unscaled infeasibilities." So I think either is correct. Let's just leave as is for now, and revisit it if we need to in the future.